src/includes/class-alma-wc-checkout-helper.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Alma cart handler
+ *
+ * @package Alma_WooCommerce_Gateway
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Not allowed' ); // Exit if accessed directly.
+}
+
+
+/**
+ * Class Alma_WC_Checkout_Helper.
+ */
+class Alma_WC_Checkout_Helper {
+	const CHECKOUT_NONCE = 'alma_checkout_nonce';
+
+	/**
+	 * The logger.
+	 *
+	 * @var Alma_WC_Logger
+	 */
+	private $logger;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->logger = new Alma_WC_Logger();
+	}
+
+	/**
+	 * Check if alma_fee_plan is set in POST data and verify wp_nonce.
+	 *
+	 * @return null|string
+	 */
+	public function get_chosen_alma_fee_plan() {
+		if ( WC()->cart === null ) {
+			wc_add_notice( '<strong>Fee plan</strong> is required.', 'error' );
+			return null;
+		}
+
+		return $this->check_nonce( 'alma_fee_plan', self::CHECKOUT_NONCE );
+	}
+
+	/**
+	 * Check if payment_method is set in POST and is an Alma payment method
+	 * then and verify wp_nonce.
+	 *
+	 * @return bool
+	 */
+	public function is_alma_payment_method() {
+		$payment_method = $this->check_nonce( 'payment_method', self::CHECKOUT_NONCE );
+		$this->logger->info( sprintf( '%s: %s', __FUNCTION__, $payment_method ) );
+		return $payment_method && substr( $payment_method, 0, 5 ) === 'alma_';
+	}
+
+	/**
+	 * Checks nonce in POST data for an Input field.
+	 *
+	 * @param string $field_name Input action name.
+	 * @param string $nonce_name Nonce name.
+	 *
+	 * @return string|null
+	 */
+	protected function check_nonce( $field_name, $nonce_name ) {
```

Comment:
Why function check_nonce() is protected and not private ? Is this class extended by another ? (I guess not)